ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKSourceFactoryTest.java

```diff
@@ -198,10 +197,59 @@ public void shouldBuildNonWindowedTable() {
     assertThat(result.getSourceStep().getSources(), is(empty()));
   }
 
+  @Test
+  public void shouldBuildV2NonWindowedTable() {
+    // Given:
+    setUpForNonWindowedTable();
+    when(ksqlConfig.getBoolean(KsqlConfig.KSQL_ROWPARTITION_ROWOFFSET_ENABLED)).thenReturn(true);
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(result, instanceOf(SchemaKTable.class));
+    assertThat(result.getSourceStep(), instanceOf(TableSource.class));
+
+    assertValidSchema(result);
+    assertThat(result.getSourceStep().getSources(), is(empty()));
+  }
+
+  @Test
+  public void shouldBuildCorrectFormatsForV2NonWindowedTable() {
+    // Given:
+    setUpForNonWindowedTable();
+    when(ksqlConfig.getBoolean(KsqlConfig.KSQL_ROWPARTITION_ROWOFFSET_ENABLED)).thenReturn(true);
+
+    // When:
+    final SchemaKStream<?> result = SchemaKSourceFactory.buildSource(
+        buildContext,
+        dataSource,
+        contextStacker
+    );
+
+    // Then:
+    assertThat(((TableSource) result.getSourceStep()).getStateStoreFormats(), is(
+        InternalFormats.of(
+            keyFormat,
+            valueFormatInfo
+        )
+    ));
+  }
+
+  private void setUpForNonWindowedTable() {
```

Comment:
It feels like there should be a naming convention for what I'm trying to do here but I didn't find it with a cursory Google search.

Reply:
The rest of the codebase uses `givenNonWindowedTable()`.

Reply:
I considered this, but the`given` methods I saw before assign a class-level variable with the table. Now looking at something like `PushQueryValidatorTest`, I see that's not always the case. I could do something like this? 

```
// Given/When:
givenNonWindowedTable()

// Then:
{assertions}
```

Or if there doesn't really need to be a strict parallel, I could just leave the semantics as is and change the variable name. WDYT?

Reply:
Sorry, I don't understand the distinction you're trying to point out. Why would it be inconsistent to rename this method `givenNonWindowedTable()`? The way you'd use it is:
```
// Given:
givenNonWindowedTable();

// When:
{statement}

// Then:
{assertions}
```